File: lib/features/dashboard/domain/entities/scan_progress.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'scan_progress.freezed.dart';
part 'scan_progress.g.dart';

enum ScanPhase {
  @JsonValue('DISK_SPACE')
  diskSpace,
  @JsonValue('PHOTOS')
  photos,
  @JsonValue('VIDEOS')
  videos,
  @JsonValue('AUDIO')
  audio,
  @JsonValue('DOCUMENTS')
  documents,
  @JsonValue('CALCULATING')
  calculating,
  @JsonValue('COMPLETE')
  complete,
  @JsonValue('PAUSED')
  paused,
  @JsonValue('CANCELLED')
  cancelled,
  @JsonValue('ERROR')
  error,
  @JsonValue('CACHE_INVALIDATED')
  cacheInvalidated,
}

@freezed
class ScanProgress with _$ScanProgress {
  const factory ScanProgress({
    required ScanPhase phase,
    required int processedItems,
    required int totalItems,
    required int currentBytes,
    String? currentVolume,
    @Default(false) bool isPaused,
  }) = _ScanProgress;

  factory ScanProgress.fromJson(Map<String, dynamic> json) =>
      _$ScanProgressFromJson(json);
}

File: android/app/src/main/kotlin/com/lebac/storage_dashboard/clear_space/models/ScanProgress.kt
package com.lebac.storage_dashboard.clear_space.models

data class ScanProgress(
    val phase: ScanPhase,
    val processedItems: Int,
    val totalItems: Int,
    val currentBytes: Long,
    val currentVolume: String? = null,
    val isPaused: Boolean = false
) {
    fun toMap(): Map<String, Any?> = mapOf(
        "phase" to phase.name,
        "processedItems" to processedItems,
        "totalItems" to totalItems,
        "currentBytes" to currentBytes,
        "currentVolume" to currentVolume,
        "isPaused" to isPaused
    )
    
    val percentage: Double
        get() = if (totalItems == 0) 0.0 else processedItems.toDouble() / totalItems
}

enum class ScanPhase {
    DISK_SPACE,
    PHOTOS,
    VIDEOS,
    AUDIO,
    DOCUMENTS,
    JUNK,
    EMPTY_FOLDERS,
    APKS,
    SIMILAR_PHOTOS,
    CALCULATING,
    COMPLETE,
    PAUSED,
    CANCELLED,
    ERROR
}
